pkg/distributor/distributor.go

```diff
@@ -696,8 +706,9 @@ func (d *Distributor) validateSeries(nowt time.Time, ts mimirpb.PreallocTimeseri
 }
 
 // wrapPushWithMiddlewares returns push function wrapped in all Distributor's middlewares.
-func (d *Distributor) wrapPushWithMiddlewares(externalMiddleware func(next push.Func) push.Func, next push.Func) push.Func {
-	var middlewares []func(push.Func) push.Func
+// externalMiddlewares will be applied in the reverse order, so the last middleware in the slice will be the outermost one.
```

Comment:
I may be wrong, but to me the code looks like `externalMiddlewares` will be applied in the order they're in, and not in reverse order?

Reply:
this iterates over the `middlewares` slice from the highest to the lowest index:

https://github.com/grafana/mimir/blob/a86cb885f9af86e19129e5a7f12fde23503646a7/pkg/distributor/distributor.go#L726-L728

Reply:
Right, but `d.limitsMiddleware` would be the first middleware to be applied to the request and the last item in `externalMiddlewares` the last (innermost), or not?

Reply:
I think you're right, thanks for pointing that out.

I'll fix the comment, I don't think it's necessary to treat the middlewares differently than they're currently treated, do you agree?

Reply:
fixed here afc9be40a5cd8b19a5d27ebcc3a0d4e12f141429

Reply:
Yes, I agree 👍